soc/nxp/rw/power.c

```diff
@@ -177,7 +178,17 @@ __weak void pm_state_set(enum pm_state state, uint8_t substate_id)
 		__WFI();
 		break;
 	case PM_STATE_SUSPEND_TO_IDLE:
+		/* save old value of main clock mux and switch to lposc */
+		uint32_t main_sel_a = CLKCTL0->MAINCLKSELA;
+		uint32_t main_sel_b = CLKCTL0->MAINCLKSELB;
+
+		CLKCTL0->MAINCLKSELA = 2;
+		CLKCTL0->MAINCLKSELB = 0;
```

Comment:
```suggestion
		CLOCK_AttachClk(kLPOSC_to_MAIN_CLK);
```
to make more readable?

Reply:
I considered this as well but that function actually has some more complexity than I expected and just wanted to do the lowest latency thing which is just these two register writes. Since the file is already SOC specific I think it's fine, since the comment already explains it's to attach LPOSC to main clock.